packages/deployer/internal/storage-ast-validator.js

```diff
@@ -0,0 +1,103 @@
+const {
+  findContractDependencies,
+  findYulStorageSlotAssignments,
+  findContractStateVariables,
+} = require('@synthetixio/core-js/utils/ast');
+const logger = require('@synthetixio/core-js/utils/logger');
+const filterValues = require('filter-values');
+
+class ModuleStorageASTValidator {
+  constructor(asts) {
+    this.asts = asts;
+  }
+
+  findDuplicateNamespaces(namespaces) {
+    const duplicates = namespaces
+      .map((s) => s.position)
+      .filter((s, index, namespaces) => namespaces.indexOf(s) !== index);
+
+    const ocurrences = [];
+
+    if (duplicates.length > 0) {
+      duplicates.map((duplicate) => {
+        const cases = namespaces.filter((s) => s.position === duplicate);
+        ocurrences.push({
+          position: duplicate,
+          contracts: cases.map((c) => c.contractName),
+        });
+      });
+    }
+
+    return ocurrences.length > 0 ? ocurrences : null;
+  }
+
+  findNamespaceCollisions() {
+    const namespaces = [];
+
+    for (var [contractName, ast] of Object.entries(this.asts)) {
+      const slot = findYulStorageSlotAssignments(contractName, ast);
+
+      if (slot) {
```

Comment:
if we always return the array on the library this `if` is not needed

Reply:
Removed